app/src/main/java/ch/epfl/sdp/map/MapsActivity.java

```diff
@@ -94,14 +97,23 @@ protected void onCreate(Bundle savedInstanceState) {
         playMode = intent.getStringExtra("playMode");
         Log.d("play mode", "The play mode: " + playMode);
 
-        Game.getInstance().setRenderer(this);
+        boolean isSolo = playMode.equals("single-player");
 
         AppContainer appContainer = ((MyApplication) getApplication()).appContainer;
         authenticationAPI = appContainer.authenticationAPI;
         commonDatabaseAPI = appContainer.commonDatabaseAPI;
         serverDatabaseAPI = appContainer.serverDatabaseAPI;
         clientDatabaseAPI = appContainer.clientDatabaseAPI;
 
+        if (Game.getInstance().gameStarted && PlayerManager.getInstance().isSoloMode() != isSolo) {
+            JunkCleaner.clearAll();

```

Comment:
If this part of code is used serveral times. Its better to make one methode to avoid code duplication.

Reply:
Which part?

Reply:
fixed

Reply:
Thanks!